x-pack/legacy/plugins/maps/public/angular/map_controller.js

```diff
@@ -367,7 +367,9 @@ app.controller(
       if (prevIndexPatternIds !== nextIndexPatternIds) {
         return;
       }
-      $scope.indexPatterns = indexPatterns;
+      $scope.$evalAsync(() => {
```

Comment:
@nreese having reverted the inequality checks, is this change alright now?

Reply:
looks good